lib/internal/modules/esm/utils.js

```diff
@@ -120,46 +118,17 @@ async function initializeHooks() {
 
 
   const { Hooks } = require('internal/modules/esm/hooks');
+  const esmLoader = require('internal/process/esm_loader').esmLoader;
+
   const hooks = new Hooks();
+  esmLoader.setDelegate(hooks);
 
-  const { DefaultModuleLoader } = require('internal/modules/esm/loader');
-  class ModuleLoader extends DefaultModuleLoader {
-    loaderType = 'internal';
-    async #getModuleJob(specifier, parentURL, importAssertions) {
-      const resolveResult = await hooks.resolve(specifier, parentURL, importAssertions);
-      return this.getJobFromResolveResult(resolveResult, parentURL, importAssertions);
-    }
-    getModuleJob(specifier, parentURL, importAssertions) {
-      const jobPromise = this.#getModuleJob(specifier, parentURL, importAssertions);
-      return {
-        run() {
-          return PromisePrototypeThen(jobPromise, (job) => job.run());
-        },
-        get modulePromise() {
-          return PromisePrototypeThen(jobPromise, (job) => job.modulePromise);
-        },
-        get linked() {
-          return PromisePrototypeThen(jobPromise, (job) => job.linked);
-        },
-      };
-    }
-    load(url, context) { return hooks.load(url, context); }
-  }
-  const privateModuleLoader = new ModuleLoader();
```

Comment:
Where did all of this code go? We don’t need it anymore?

Reply:
> Where did all of this code go?

The re-useable parts live in `DefaultModuleLoader` now: https://github.com/nodejs/node/pull/48559/files#diff-f6796082f599554ec3a29c47cf026cb24fc5104884f2632e472c05fe622d778bR150. 

> We don’t need it anymore?

Since the base `DefaultModuleLoader` handles this machinery the "delegates" don't need to.

Reply:
Ah cool. I wondered about this when I was working on my PR. Glad that is the case! 😁